File: Frontend/src/app/models/post.ts
import { Comment_ } from './comment';

export class Post
{
    constructor
    (
        public id: number,
        public body: string,
        public dateTime: Date,
        public comments: Comment_[]    
    )
    {}

    static newPost(body: string, comments?: Comment_[]): Post {
        return new Post(0, body, new Date(), comments || []);
    }
    
}

File: Backend/Entity_Layer/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity_Layer
{
    public class Post : Parent
    {
        
        [Required]
        [MinLength(1)]
        public string Body { get; set; }

        public IEnumerable<Comment> Comments { get; set; }
    }
}
